crates/futures/src/lib.rs

```diff
@@ -50,18 +50,40 @@ pub use js_sys;
 pub use wasm_bindgen;
 
 mod task {
+    use std::future::Future;
+    use std::pin::Pin;
+
     use cfg_if::cfg_if;
 
-    cfg_if! {
-        if #[cfg(target_feature = "atomics")] {
-            mod wait_async_polyfill;
-            mod multithread;
-            pub(crate) use multithread::*;
+    #[cfg(target_feature = "atomics")]
+    mod multithread;
+    mod singlethread;
+    #[cfg(target_feature = "atomics")]
+    mod wait_async_polyfill;
+
+    pub(crate) fn spawn(future: Pin<Box<dyn Future<Output = ()> + 'static>>) {
+        cfg_if! {
+            if #[cfg(target_feature = "atomics")] {
+                #[wasm_bindgen::prelude::wasm_bindgen]
+                extern "C" {
+                    /// Returns [`crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated) global property.
+                    #[wasm_bindgen(js_name = crossOriginIsolated)]
+                    static CROSS_ORIGIN_ISOLATED: bool;
```

Comment:
This only exists inside browsers, so it'll break if you try to use it inside Node or Deno.

You should be able to check if it exists at runtime, and then if it doesn't we can conservatively assume that multithreading is possible.